art_internal.hpp

```diff
@@ -31,19 +31,51 @@ class [[nodiscard]] basic_db_leaf_deleter;
 // Internal ART key in binary-comparable format
 template <typename KeyType>
 struct [[nodiscard]] basic_art_key final {
+
+  // Convert an external key into an internal key supporting
+  // lexicographic comparison.  This is only intended for key types
+  // for which simple conversions are possible.  For complex keys,
+  // including multiple key components or Unicode data, the
+  // application should use a gsl::space<std::byte> which already
+  // supports lexicographic comparison.
   [[nodiscard, gnu::const]] static UNODB_DETAIL_CONSTEXPR_NOT_MSVC KeyType
   make_binary_comparable(KeyType key) noexcept;
 
+  // Convert an internal key into an external key. This is only
+  // intended for key types for which simple conversions are possible.
+  // For complex keys, including multiple key components or Unicode
+  // data, the application should use a gsl::space<std::byte> which
+  // already supports lexicographic comparison.
+  [[nodiscard, gnu::const]] static UNODB_DETAIL_CONSTEXPR_NOT_MSVC KeyType
+  make_external(KeyType key) noexcept;
+
   constexpr basic_art_key() noexcept = default;
 
   UNODB_DETAIL_CONSTEXPR_NOT_MSVC explicit basic_art_key(KeyType key_) noexcept
       : key{make_binary_comparable(key_)} {}
 
+  // @return true iff the two keys are equal.
```

Comment:
IMHO a redundant comment, given that this is `bool operator ==`

Reply:
Well.  There is always reading the code :-).  I tend to prefer a common on each class and method.  And, yes, sometimes it is beside the point.  I will remove it.